catalogue/models/component/__init__.py

```diff
@@ -202,3 +209,24 @@ def get_technical_specification(self) -> Tuple[str, List[str]]:
 
     def get_source(self) -> Tuple[str, List[str]]:
         return Source._meta.verbose_name, [x.name for x in Source._meta.get_fields()]
+
+@receiver(signals.pre_save, sender=Component)
+def init_unapproved_state(sender, instance, *args, **kwargs):
+    previous = Component.objects.get(id=instance.id)
```

Comment:
error here when component not exists (i.e when creating the component). Will fix it